core/src/main/java/org/apache/calcite/sql/type/ReturnTypes.java

```diff
@@ -551,6 +551,13 @@ public static SqlCall stripSeparator(SqlCall call) {
       opBinding -> opBinding.getTypeFactory().leastRestrictive(
           opBinding.collectOperandTypes());
 
+  public static final SqlReturnTypeInference NVL2_RESTRICTIVE = opBinding -> {
```

Comment:
NEed javadoc

Reply:
done

Reply:
This looks wrong. If the first operand does not have the type null then you need to unify the types of operands 2 and 3. In fact, I think you always need to unify them.

Reply:
@mihaibudiu You mean I need to uniformly convert to biggesttype?